packages/rest/src/http-handler.ts

```diff
@@ -58,30 +49,23 @@ export class HttpHandler {
     return this._routes.describeApiPaths();
   }
 
-  findRoute(request: ParsedRequest): ResolvedRoute {
+  findRoute(request: Request): ResolvedRoute {
     return this._routes.find(request);
   }
 
-  protected async _handleRequest(
-    request: ServerRequest,
-    response: ServerResponse,
-  ): Promise<void> {
-    const parsedRequest: ParsedRequest = parseRequestUrl(request);
-    const requestContext = this._createRequestContext(parsedRequest, response);
+  async handleRequest(httpCtx: HttpContext): Promise<void> {
+    const requestContext = this._createRequestContext(httpCtx);
 
     const sequence = await requestContext.get<SequenceHandler>(
       RestBindings.SEQUENCE,
     );
-    await sequence.handle(parsedRequest, response);
+    await sequence.handle(httpCtx);
   }
 
-  protected _createRequestContext(
-    req: ParsedRequest,
-    res: ServerResponse,
-  ): Context {
+  protected _createRequestContext(httpCtx: HttpContext): Context {
     const requestContext = new Context(this._rootContext);
-    requestContext.bind(RestBindings.Http.REQUEST).to(req);
-    requestContext.bind(RestBindings.Http.RESPONSE).to(res);
+    requestContext.bind(RestBindings.Http.REQUEST).to(httpCtx.request);
+    requestContext.bind(RestBindings.Http.RESPONSE).to(httpCtx.response);
```

Comment:
I think we should replace request/response bindings with a single binding holding the full HTTP context. Thoughts?